sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -206,10 +206,14 @@ public void Dispose()
             if (response != null)
             {
                 _response = null;
-                response.Dispose();
+                // ================================================================
```

Comment:
Is this an illustration of an approach to work around the issue folks are seeing when they override the transport?

For that, I was wondering if we could use a special internal MemoryStream to buffer the response.  I think the problem is that we're disposing the response content stream if/when the stream type is MemoryStream, when what we mean is "if we have buffered the stream".  My hypothesis is that using a custom subtype of MemoryStream would let us do a more precise check, and not dispose a MemoryStream coming from a custom transport.

Would you be interested in exploring that option?